TheRealReal/heroku-review-envs
# [#169683271] Write Script that Flips an review environment to .ninja or .com

[PT-#169683271](https://www.pivotaltracker.com/story/show/169683271)

## Description
This PR adds a new job to our `heroku-review-envs` scripts. 

### heroku-config-var-set
This job allows you to set any number of config variables on a specific heroku application. When calling it you give it the Heroku API token as an environment variable, then pass in a number of arguments specifying the heroku app you want to set it them on and the config variables you want to set. 

I followed the same pattern that was present in the `APP_REF` for specifying config variables since passing raw JSON did not work well with github workflows. You must specify your CONFIG_VARS like so: 

`CONFIG_VARS=var1%value1|var2%value2|var3%value3`

`%` is used in place of `=`
`|` is used to separate variables

Usage: 
**build**
`docker build -t config-var-set heroku-config-var-set/`

**run**
`docker run -e HEROKU_API_TOKEN={Heroku Token} config-var-set PP_ORIGIN=web APP_TARGET=web APP_PREFIX=trr CONFIG_VARS=OKTA_AUTH_SERVER_BASE_URL%https://login.therealreal.com/oauth2/default/|OKTA_CLIENT_ID%0oa18r3unfqJuF9Y4357|OKTA_CLIENT_SECRET%{Client Secret}`

## Motivation and Context
We want to be able to easily flip between .com and .ninja Okta settings for review environments. These jobs will facilitate that by adding a label to the PR that controls the arguments that are passed into the above jobs. 

## Testing
- For testing purposes you can use the following PR: https://github.com/TheRealReal/real-server/pull/16711
- If you add the `okta-ninja` label, or remove it, it should adjust the Okta envrionment that the review env is configured for
- Make sure all checks are finished running before you attempt to test the adjusted settings
- Visit the following URL with a brand new incognito browser for each test: https://trr-web-pr-16711.herokuapp.com/admin/okta
- If you've configured it for .com, you should be able to use your own .com account to login
- If you've configured it for .ninja, use one of the test users like `price.administrator@therealreal.ninja`
- Nothing should be tested much further than login, that is all that is affected by this. The Okta role groups should work as expected, however, so price.administrator should have price administration privileges, for example. 

## Type of Changes



## Checklist:
